File: src/main/java/me/devsnox/battlekits/commands/ShardCommand.java
package me.devsnox.battlekits.commands;

import me.devsnox.battlekits.KitManager;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class ShardCommand implements CommandExecutor {

    private KitManager kitManager;

    public ShardCommand(KitManager kitManager) {
        this.kitManager = kitManager;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if(sender instanceof Player) {
            Player player = ((Player) sender).getPlayer();

            if(player.hasPermission("battlekits.getshard")) {
                if(args.length == 1) {
                    try {
                        player.getInventory().addItem(this.kitManager.getShard(Integer.valueOf(args[0])));
                        return true;
                    } catch (Exception ex) {
                    }
                }

                player.sendMessage("§cBenutze /" + command.getName() + " <id>");
            }
        }
        return false;
    }
}

File: spigot/src/main/kotlin/me/devsnox/battlekits/commands/ShardCommand.kt
package me.devsnox.battlekits.commands

import me.devsnox.battlekits.KitManager
import me.devsnox.battlekits.functions.javaPlugin
import net.darkdevelopers.darkbedrock.darkness.spigot.commands.Command
import net.darkdevelopers.darkbedrock.darkness.spigot.functions.sendTo
import net.darkdevelopers.darkbedrock.darkness.spigot.utils.isPlayer
import org.bukkit.ChatColor
import org.bukkit.command.CommandSender

class ShardCommand(private val kitManager: KitManager) : Command(
    javaPlugin,
    "Shard",
    "battlekits.getshard",
    usage = "<id>",
    minLength = 1,
    maxLength = 1
) {
    override fun perform(sender: CommandSender, args: Array<String>): Unit = sender.isPlayer { player ->
        val id = args[0].toIntOrNull()
        if (id == null) "${ChatColor.RED}Gib eine Ganzzahl an!".sendTo(sender)
        else player.inventory.addItem(kitManager.getShard(id))
    }
}
